runtime/logging.go

```diff
@@ -73,8 +74,26 @@ func (h *LoggingHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				bs, r.Body, err = readBody(r.Body)
 			}
 			if err == nil {
-				fields["req_body"] = string(bs)
-			} else {
+				if gzipReceived(r.Header) {
```

Comment:
It's somewhat unfortunate that we have to decompress what we've ourselves compressed before just for logging it. I wonder if there's a different way of chaining handlers possible that would let us avoid that...